src/org/zaproxy/zap/extension/replacer/ExtensionReplacer.java

```diff
@@ -0,0 +1,263 @@
+                Pattern p = null;
+                if (rule.isMatchRegex()) {
+                    p = Pattern.compile(rule.getMatchString());
+                }
+                switch (rule.getMatchType()) {
+                case req_header:
+                    LOGGER.debug("Add in request header: " + rule.getMatchString() + " : " + rule.getReplacement());
+                    msg.getRequestHeader().setHeader(rule.getMatchString(), rule.getReplacement());
+                    break;
+                case req_header_str:
+                    LOGGER.debug("Replace in request header: " + rule.getMatchString() + " with " + rule.getReplacement());
+                    String header = msg.getRequestHeader().toString();
+                    if (contains(header, rule.getMatchString(), p)) {
+                        header = replace(header, rule.getMatchString(), p, rule.getReplacement());
+                        try {
+                            msg.setRequestHeader(new HttpRequestHeader(header));
+                        } catch (HttpMalformedHeaderException e) {
+                            LOGGER.error(e.getMessage(), e);
+                        }
+                    }
+                    break;
+                case req_body_str:
+                    LOGGER.debug("Add in request body: " + rule.getMatchString() + " : " + rule.getReplacement());
+                    String body = msg.getRequestBody().toString();
+                    if (contains(body, rule.getMatchString(), p)) {
+                        body = replace(body, rule.getMatchString(), p, rule.getReplacement());
+                        msg.getRequestBody().setBody(body);
+                        msg.getRequestHeader().setContentLength(msg.getRequestBody().length());
+                    }
+                    break;
+                case resp_header:
+                    // Ignore response rules here
+                    LOGGER.debug("Ignore response rule " + rule.getDescription());
+                    break;
+                case resp_header_str:
+                    // Ignore response rules here
+                    LOGGER.debug("Ignore response rule " + rule.getDescription());
+                    break;
+                case resp_body_str:
+                    // Ignore response rules here
+                    LOGGER.debug("Ignore response rule " + rule.getDescription());
+                    break;
+                }
+            }
+        }
+    }
+
+    @Override
+    public void onHttpResponseReceive(HttpMessage msg, int initiator, HttpSender httpSender) {
+        for (ReplacerParamRule rule : this.getParams().getRules()) {
+            Pattern p = null;
+            if (rule.isMatchRegex()) {
+                p = Pattern.compile(rule.getMatchString());
+            }
+            if (rule.isEnabled() && rule.appliesToInitiator(initiator)) {
+                switch (rule.getMatchType()) {
+                case req_header:

```

Comment:
This and following case could be "merged" with `req_body_str`, it's doing the same thing.